cmd/trace-agent/config/config.go

```diff
@@ -597,3 +594,33 @@ func acquireHostnameFallback(c *config.AgentConfig) error {
 	log.Debugf("Acquired hostname from core agent (%s): %q.", c.DDAgentBin, c.Hostname)
 	return nil
 }
+
+// UpdateConfigHandler returns handler for runtime configuration changes.
+func UpdateConfigHandler() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		_ = req.ParseForm()
+		setting := html.UnescapeString(req.Form.Get("setting"))
+		value := html.UnescapeString(req.Form.Get("value"))
```

Comment:
This `setting` and `value` format, is it something that was universally decided by means of an RFC or does it mirror some other agent's behaviour?

If not, I think we can do better here given that in the future we may want to change more than one setting, perhaps even in the same request, so why not something like this:

```
curl -X POST http://localhost:8126/config/set?log_level=warn&apm_config.receiver_timeout=2
```

What do you think?

Obviously for now we only have `log_level`, but judging by the name of the path (`/config/set`) it seems like more might come (otherwise why not call it `/config/log_level?v=warn`).

Reply:
Unless I change mux Router to gorilla or alike, no, because Go’s servemux is pretty lightweight and doesn’t support such routing

Reply:
What do you mean? You can do all of the above with `net/http`. The request's `URL` field has the query in there which you can parse with [`url.ParseQuery`](https://pkg.go.dev/net/url#ParseQuery)

Reply:
You can do that too. But I think we don't need to :)